ai/core/src/unitycatalog/ai/core/databricks.py

```diff
@@ -250,16 +231,15 @@ def _is_spark_session_active(self):
             return not self.spark.is_stopped
         return self.spark.getActiveSession() is not None
 
-    def set_default_spark_session(self):
-        if not self._is_spark_session_active():
-            _validate_databricks_connect_available()
-            from databricks.connect.session import DatabricksSession as SparkSession
+    def set_spark_session(self):
+        _validate_databricks_connect_available()
```

Comment:
I think we still need `not self._is_spark_session_active()` check, otherwise this will run everytime inside create_function. But the check is not needed in `refresh_client_and_session` though, so probably we need another function like `set_spark_session_if_stopped`

Reply:
added the inheritance chain in order for us to not continually hammer that method :)